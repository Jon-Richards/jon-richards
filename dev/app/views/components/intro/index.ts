/**
 * @fileoverview
 * Entrypoint for the Intro module.
 */

export { Intro, IntroProps } from './intro';
