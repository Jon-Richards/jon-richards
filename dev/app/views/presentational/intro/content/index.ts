export { Content, ContentProps } from './content';
