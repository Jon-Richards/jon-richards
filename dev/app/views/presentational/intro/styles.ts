import { Breakpoints } from 'Config/styles/breakpoints';

/** Styles for the Intro component. */
export const STYLES = {
  root: `
    display: flex;
    position: relative;
    width: 100%;
    height: auto;
    min-height: 100vh;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
  `,
  wrapper: `
    display: flex;
    position: relative;
    z-index: 1;
    width: auto;
    margin: auto;
    max-width: 45rem;
    height: auto;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    @media( min-width: ${Breakpoints[480]}) {
      margin: auto 4rem;
    }

    @media (min-width: ${Breakpoints[720]}) {
      display: inline-flex;
      flex-shrink: 1;
      flex-direction: row;
      align-items: flex-start;
    }
  `
};
