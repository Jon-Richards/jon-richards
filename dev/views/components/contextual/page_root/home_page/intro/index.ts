/**
 * @fileoverview
 * Entrypoint for the Intro module.
 */

export { Intro } from './intro';
