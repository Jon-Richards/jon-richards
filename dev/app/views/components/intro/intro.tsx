/**
 * @fileoverview
 * Parent component to the intro section.
 */

import * as React from 'react';
import { v4 as uuid } from 'uuid';
import { SkillFilters, SkillFilter } from './content/skill_filters';
import { Panel } from 'Views/components/panel';
import { IntroContent } from './content';
const CSS = require('./intro.scss');

/** Props needed to instantiate the Intro component. */
type Props = {
  /** The intro title. */
  title: React.ReactFragment;
  /** The subtitle. */
  subtitle: React.ReactFragment;
  /** Array of skills by which portfolio pieces can be ordered. */
  skills?: SkillFilter[];
  /** An optional theme with qhich to render the intro.  Defaults to "PANEL" */
  theme?: 'PANEL' | 'COPY';
};

/** Renders the introduction section. */
function Intro (props: Props): JSX.Element {
  const {
    title,
    subtitle,
    skills = [],
    theme = 'PANEL',
  } = props;

  const mockSkills: SkillFilter[] = [
    {
      uuid: uuid(),
      displayLabel: 'TypeScript',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'JavaScript',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Node',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'CSS / SASS',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'SVG',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'PHP',
      value: 'typescript',
      isActive: false,
      isCore: false,
    }
  ];

  return (
    <div className={CSS['root']}>
      <div className={CSS['wrapper']}>
        {theme === 'PANEL' ? (
          <Panel>
            <IntroContent
              title={title} 
              subtitle={subtitle} 
              skills={mockSkills}
            />
          </Panel>
        ) : (
          <IntroContent title={title} subtitle={subtitle} skills={mockSkills} />
        )}
      </div>
    </div>
  );
}

const introMemo = React.memo<Props>(Intro);

export { introMemo as Intro, Props as IntroProps };