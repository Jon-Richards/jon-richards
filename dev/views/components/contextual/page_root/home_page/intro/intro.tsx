/**
 * @fileoverview
 * Parent component to the intro section.
 */

import * as React from 'react';
import { v4 as uuid } from 'uuid';
import { SkillFilters, SkillFilter } from './content/skill_filters';
import { Panel } from '../../../../shared/panel';
import { Content } from './content';
const CSS = require('./intro.scss');

type Props = {
  /** The intro title. */
  title: React.ReactFragment;
  /** The subtitle. */
  subtitle: React.ReactFragment;
  /** Array of skills by which portfolio pieces can be ordered. */
  skills?: SkillFilter[];
};

/** Renders the introduction section. */
function Intro (props: Props): JSX.Element {
  const { title, subtitle, skills = [] } = props;

  const mockSkills: SkillFilter[] = [
    {
      uuid: uuid(),
      displayLabel: 'TypeScript',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'JavaScript',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Node',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'CSS / SASS',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'SVG',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'PHP',
      value: 'typescript',
      isActive: false,
      isCore: false,
    }
  ];

  let renderWithPanel = false;

  if (window) {
    if (window.matchMedia('(min-width: 480px)').matches === true) {
      renderWithPanel = true;
    }
  }

  return (
    <div className={CSS['root']}>
      <div className={CSS['wrapper']}>
        {renderWithPanel ? (
          <Panel display="flex">
            <Content title={title} subtitle={subtitle} skills={mockSkills} />
          </Panel>
        ) : (
          <Content title={title} subtitle={subtitle} skills={mockSkills} />
        )}
      </div>
    </div>
  );
}

const introMemo = React.memo<Props>(Intro);

export { introMemo as Intro };