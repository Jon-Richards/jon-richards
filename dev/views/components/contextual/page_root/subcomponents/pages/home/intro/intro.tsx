/**
 * @fileoverview
 * Parent component to the intro section.
 */

import * as React from 'react';
import { v4 as uuid } from 'uuid';
import { SkillFilters, SkillFilter } from './content/skill_filters';
import { Panel } from '../../../../../../shared/panel';
import { Content } from './content';
const CSS = require('./intro.scss');

/** Parent component for the intro section. */
export const INTRO = React.memo<{
  /** The intro title. */
  title: React.ReactFragment;
  /** The subtitle. */
  subtitle: React.ReactFragment;
  /** Array of skills by which portfolio pieces can be ordered. */
  skills?: SkillFilter[];
}>(props => {
  const { title, subtitle, skills = [] } = props;

  const mockSkills: SkillFilter[] = [
    {
      uuid: uuid(),
      displayLabel: 'TypeScript',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'JavaScript',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Node',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'CSS / SASS',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'SVG',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'PHP',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Rails',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Photoshop',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'Illustrator',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'SVG',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'PHP',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Rails',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Photoshop',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'Illustrator',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'SVG',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'PHP',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Rails',
      value: 'typescript',
      isActive: false,
      isCore: false,
    },
    {
      uuid: uuid(),
      displayLabel: 'Photoshop',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
    {
      uuid: uuid(),
      displayLabel: 'Illustrator',
      value: 'typescript',
      isActive: false,
      isCore: true,
    },
  ];

  let renderWithPanel = false;

  if (window) {
    if (window.matchMedia('(min-width: 480px)').matches === true) {
      renderWithPanel = true;
    }
  }

  return (
    <div className={CSS['root']}>
      <div className={CSS['wrapper']}>
        {renderWithPanel ? (
          <Panel display="flex">
            <Content title={title} subtitle={subtitle} skills={mockSkills} />
          </Panel>
        ) : (
          <Content title={title} subtitle={subtitle} skills={mockSkills} />
        )}
      </div>
    </div>
  );
});
