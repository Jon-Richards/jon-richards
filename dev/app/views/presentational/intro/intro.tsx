/** @jsx jsx */

import * as React from 'react';
import { Panel } from 'Views/presentational/panel';
import { Content, ContentProps } from './content';
import { css, jsx } from '@emotion/core';
import { STYLES } from './styles';

/** Props needed to instantiate the Intro component. */
type Props = {
  /** The intro title. */
  title: React.ReactFragment;
  /** The subtitle. */
  subtitle: React.ReactFragment;
  /** An array of skills. */
  skills: ContentProps['skills'];
  /** An optional theme with which to render the intro.  Defaults to "PANEL" */
  theme?: 'PANEL' | 'COPY';
};

/** Renders the introduction section. */
function Intro (props: Props): JSX.Element {
  const {
    title,
    subtitle,
    skills,
    theme = 'PANEL',
  } = props;

  return (
    <div css={css(STYLES.root)}>
      <div css={css(STYLES.wrapper)}>
        {theme === 'PANEL' ? (
          <Panel data-testid="intro_panel">
            <Content title={title} subtitle={subtitle} skills={skills} />
          </Panel>
        ) : (
          <Content title={title} subtitle={subtitle} skills={skills} />
        )}
      </div>
    </div>
  );
}

const introMemo = React.memo<Props>(Intro);

export { introMemo as Intro, Props as IntroProps };