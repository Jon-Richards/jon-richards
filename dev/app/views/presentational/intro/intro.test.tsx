import * as React from 'react';
import { mount, render } from 'enzyme';
import { Intro } from './intro';

describe('The Intro component.', () => {
  it('Should match the snapshot.', () => {
    const test_1 = render(
      <Intro title="Foo" subtitle="Bar" theme="PANEL" skills={[]} />
    );

    expect(test_1).toMatchSnapshot();
  });

  it('Should render according to its props.', () => {
    const test_1 = mount(
      <Intro title="Foo" subtitle="Bar" theme="PANEL" skills={[]} />
    );
    const test_2 = render(
      <Intro title="Apple" subtitle="Banana" theme="COPY" skills={[]} />
    );

    expect(test_1.find('[data-testid="intro_panel"]').length).toBe(1);
    expect(test_2.find('[data-testid="intro_panel"]').length).toBe(0);
  });
});
