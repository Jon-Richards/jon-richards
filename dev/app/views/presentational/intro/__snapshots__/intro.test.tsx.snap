// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Intro component. Should match the snapshot. 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: 100%;
  height: auto;
  min-height: 100vh;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  z-index: 1;
  width: auto;
  margin: auto;
  max-width: 45rem;
  height: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media ( min-width:480px) {
  .emotion-6 {
    margin: auto 4rem;
  }
}

@media (min-width:720px) {
  .emotion-6 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    -webkit-flex-shrink: 1;
    -ms-flex-negative: 1;
    flex-shrink: 1;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 100%;
  height: auto;
  position: relative;
  padding: 1px;
  margin: 0rem;
  overflow: hidden;
  background-color: hsla(0,0%,14%,1);
  background-image: linear-gradient( 170deg, hsla(0,0%,30%,1), hsla(0,0%,10%,1), hsla(0,0%,17%,1) );
  box-shadow: 0.4rem 0.8rem 2rem 0rem hsla(0,0%,0%,0.5);
}

.emotion-4 {
  display: block;
  width: 100%;
  height: 100%;
  margin: 0rem;
  padding: 0rem;
  background-color: hsla(0,0%,14%,1);
  background-image: linear-gradient( 170deg, hsla(0,0%,14%,1), hsla(0,0%,8%,1) );
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem;
}

@media (min-width:720px) {
  .emotion-3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

@media (min-width:480px) {
  .emotion-2 {
    width: auto;
    margin: auto;
  }
}

@media (min-width:720px) {
  .emotion-2 {
    width: auto;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.2rem;
  font-family: "Heebo",sans-serif;
  font-size: 2.8rem;
  color: hsla(48,7%,86%,1);
  line-height: 3.2rem;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 400;
  font-size: 1.4rem;
  color: hsla(48,7%,86%,1);
}

<div
  class="emotion-7"
>
  <div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <div
        class="emotion-4"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-2"
          >
            <h1
              class="emotion-0"
            >
              Foo
            </h1>
            <h2
              class="emotion-1"
            >
              Bar
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
